migration/20170224-copy-collection-configuration-into-database.py

```diff
@@ -0,0 +1,119 @@
+import sys
+from pdb import set_trace
+bin_dir = os.path.split(__file__)[0]
+package_dir = os.path.join(bin_dir, u"..")
+sys.path.append(os.path.abspath(package_dir))
+from api.config import Configuration
+from core.model import (
+    get_one_or_create,
+    production_session,
+    Library,
+    Collection,
+)
+
+# We're going directly against the configuration object, rather than
+# using shortcuts like OverdriveAPI.from_environment, because this
+# script may be running against a branch where the implementation of
+# those shortcuts goes against the database.
+
+_db = production_session()
+
+def copy_library_registry_information(_db, library):
+    config = Configuration.integration("Adobe Vendor ID")
+    if not config:
+        print u"No Adobe Vendor ID configuration, not setting short name or secret."
+    library.library_registry_short_name = config.get("library_short_name")
+    library.library_registry_shared_secret = config.get("authdata_secret")
+
+
+def convert_overdrive(_db, library):
+    config = Configuration.integration('Overdrive')
+    if not config:
+        print u"No Overdrive configuration, not creating a Collection for it."
+    print u"Creating Collection object for Overdrive collection."
+    username = config.get('client_key')
+    password = config.get('client_secret')
+    library_id = config.get('library_id')
+    website_id = config.get('website_id')
+
+    collection, ignore = get_one_or_create(
+        _db, Collection,
+        protocol=Collection.OVERDRIVE,
+        name="Overdrive"
+    )
+    library.collections.append(collection)
+    collection.username = username
+    collection.password = password
+    collection.external_account_id = library_id
+    collection.set_setting("website_id", website_id)
+
+def convert_bibliotheca(_db, library):
+    config = Configuration.integration('3M')
+    if not config:
+        print u"No Bibliotheca configuration, not creating a Collection for it."
+        return
+    print u"Creating Collection object for Bibliotheca collection."
+    username = config.get('account_id')
+    password = config.get('account_key')
+    library_id = config.get('library_id')
+    collection, ignore = get_one_or_create(
+        _db, Collection,
+        protocol=Collection.BIBLIOTHECA,
+        name="Bibliotheca"
+    )
+    library.collections.append(collection)
+    collection.username = username
+    collection.password = password
+    collection.external_account_id = library_id
+
+def convert_axis(_db, library):
+    config = Configuration.integration('Axis 360')
+    if not config:
+        print u"No Axis 360 configuration, not creating a Collection for it."
+        return
+    print u"Creating Collection object for Axis 360 collection."
+    username = config.get('username')
+    password = config.get('password')
+    library_id = config.get('library_id')
+    # This is not technically a URL, it's u"production" or u"staging",
+    # but it's converted into a URL internally.
```

Comment:
I was imagining in the admin interface, if someone needs to change from production to staging and back to test something, it would be better to change the nicknames rather than have to remember what the URLs are.